main_test.go

```diff
@@ -329,14 +329,30 @@ func Test_validateBaseURL(t *testing.T) {
 	}
 }
 
+func TestPrepareServerDefaults(t *testing.T) {
+	testServer := prepareServer("", 8080)
+
+	if testServer.Addr != ":8080" {
+		t.Errorf("host %s and port %d != %s", "", 8080, testServer.Addr)
+	}
+}
+
+func TestPrepareServerOptions(t *testing.T) {
+	testServer := prepareServer("1.2.3.4", 5678)
+
+	if testServer.Addr != "1.2.3.4:5678" {
+		t.Errorf("host %s and port %d != %s", "1234", 5678, testServer.Addr)
+	}
+}
+
 func Test_preparedServer(t *testing.T) {
 	// For the purpose of testing - ensure we're using IPv4.
 	conf.BaseURL = "http://127.0.0.1:8000/"
 
 	testServer := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintln(w, "Hello, client")
 	}))
-	testServer.Config = prepareServer()
+	testServer.Config = prepareServer("", 8080)
```

Comment:
Pass 0 in here so that we don't need to leave port 8080 open just to run the unit tests.

Reply:
It might be that httptest ignores the Addr setting. I can't recall